packages/frontend/core/src/components/cloud/share-header-right-item/authenticated-item.tsx

```diff
@@ -1,24 +1,39 @@
 import { Button } from '@affine/component/ui/button';
+import { useCurrentUser } from '@affine/core/hooks/affine/use-current-user';
+import { useMembers } from '@affine/core/hooks/affine/use-members';
+import { useNavigateHelper } from '@affine/core/hooks/use-navigate-helper';
 import { useAFFiNEI18N } from '@affine/i18n/hooks';
+import { useCallback, useEffect } from 'react';
 
-import { useCurrentUser } from '../../../hooks/affine/use-current-user';
-import { useMembers } from '../../../hooks/affine/use-members';
-import { useNavigateHelper } from '../../../hooks/use-navigate-helper';
 import type { ShareHeaderRightItemProps } from './index';
+import * as styles from './styles.css';
 
-export const AuthenticatedItem = ({ ...props }: ShareHeaderRightItemProps) => {
-  const { workspaceId, pageId } = props;
+export const AuthenticatedItem = ({
+  setShowDivider,
+  ...props
+}: { setShowDivider: (show: boolean) => void } & ShareHeaderRightItemProps) => {
+  const { workspaceId, pageId, publishMode } = props;
   const user = useCurrentUser();
   const members = useMembers(workspaceId, 0);
   const isMember = members.some(m => m.id === user.id);
   const t = useAFFiNEI18N();
   const { jumpToPage } = useNavigateHelper();
 
+  const handleEdit = useCallback(() => {
+    jumpToPage(workspaceId, pageId);
+  }, [workspaceId, pageId, jumpToPage]);
+
+  useEffect(() => {
```

Comment:
I think we can get this value in parent component so we don't need this effect. Is there any reason that I've not considered why it's necessary?

Reply:
Because `useMember` sends a request to the database and must be `authenticated` to be used, `useMember` cannot extract to its parent component.